simd/unit_tests/TestSIMD.cpp

```diff
@@ -277,6 +295,25 @@ class divides {
   }
 };
 
+class absolutes {
+ public:
+  template <typename T>
+  auto on_host(T const& a) const {
+    if constexpr (std::is_arithmetic_v<T>) {
+      if constexpr (std::is_signed_v<T>)
+        return std::abs(a);
+      else
+        return a;
+    } else {
+      return Kokkos::Experimental::abs(a);
+    }
+  }
+  template <typename T>
+  KOKKOS_INLINE_FUNCTION auto on_device(T const& a) const {
+    return Kokkos::Experimental::abs(a);
```

Comment:
Where is this called from?

Reply:
Unit test for device side added in [c5dc8c9](https://github.com/kokkos/kokkos/pull/6069/commits/c5dc8c99209b6e15439a32dc64fbccdb1156d23a).
 Separated each `on_host` and `on_device` to two functions to avoid the known `missing return statement` warning problem with `if constexpr` on cuda build.